modules/http-api/src/main/java/org/mule/service/http/api/domain/message/HttpMessageBuilder.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.http.api.domain.message;
+
+import static java.util.Optional.ofNullable;
+import static org.mule.runtime.api.util.Preconditions.checkNotNull;
+import org.mule.service.http.api.domain.CaseInsensitiveParameterMap;
+import org.mule.service.http.api.domain.ParameterMap;
+import org.mule.service.http.api.domain.entity.EmptyHttpEntity;
+import org.mule.service.http.api.domain.entity.HttpEntity;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public abstract class HttpMessageBuilder<B extends HttpMessageBuilder, M extends HttpMessage> {
+
+  protected ParameterMap headers = new CaseInsensitiveParameterMap();
+  protected HttpEntity entity = new EmptyHttpEntity();
+
+  /**
+   * @param entity the {@link HttpEntity} that should be used as body for the {@link HttpMessage}. Non null.
+   * @return this builder
+   */
+  public B setEntity(HttpEntity entity) {
+    checkNotNull(entity, "entity cannot be null, use an EmptyHttpEntity instead");
```

Comment:
In case if the client of this build doesn't call this method, is still needed to set an EmptyHttpEntity?